src/osd/OSD.h

```diff
@@ -1909,6 +1909,8 @@ class OSD : public Dispatcher,
   // -- placement groups --
   RWLock pg_map_lock; // this lock orders *above* individual PG _locks
   ceph::unordered_map<spg_t, PG*> pg_map; // protected by pg_map lock
+  std::set<pg_t> pending_creates_from_osd;
+  unsigned pending_creates_from_mon = 0;
```

Comment:
It looks like these 2 fields are protected by a *read* lock on pg_map_lock.. but that's not sufficient, right?  Is there some other lock happening (e.g., all in osd_lock paths)?  Maybe it's better to protect these members by their own lock?

Reply:
ahh, my bad! it's apparently wrong to use a shared lock here. fixed and repushed.